### ISSUE TITLE
[Lib] remove ` match .. | _ ->  .. `

### ISSUE THREAD

**Issue by ACoquereau** [OCamlPro/alt-ergo#295]

**Comment by Halbaroth** [comment]
I guess that this issue was about code style. Close in favor of #967

**Comment by bclement-ocp** [comment]
I don't think this is about code style, strictly speaking. We have many places with wildcard `_` in matches that have caused issues in the past — we should enable the "fragile-match" warning and remove most of them, which is I think what this issue is about. I'd rather not mix it with the inevitable bikeshedding on code style so I'll reopen.
